### PR TITLE
Add support for slash-command groups on cogs

### PR DESCRIPTION
**Pull request by HexF**
## Summary

This PR adds support for slash-command groups on cogs

An example has been provided, also demoing that permissions work

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/commands/commands.py [hunk 1]
```diff
@@ -848,6 +852,49 @@ async def invoke_autocomplete_callback(self, ctx: AutocompleteContext) -> None:
...
+
+        Returns
+        --------
+        :class:`SlashCommandGroup`
+            A new instance of this command.
+        """
+        ret = self.__class__(
+            self.name,
+            self.description,
+            **self.__original_kwargs__)
```
Comment (Dorukyum) [inline discord/commands/commands.py:866]:
```suggestion
            **self.__original_kwargs__,
        )
```

File: discord/commands/commands.py [hunk 2]
```diff
@@ -771,6 +774,7 @@ def to_dict(self) -> Dict:
             "name": self.name,
             "description": self.description,
             "options": [c.to_dict() for c in self.subcommands],
+            "default_permission": self.default_permission
```
Comment (Dorukyum) [inline discord/commands/commands.py:777]:
```suggestion
            "default_permission": self.default_permission,
```

File: discord/cog.py [hunk 3]
```diff
@@ -145,6 +145,13 @@ def __new__(cls: Type[CogMeta], *args: Any, **kwargs: Any) -> CogMeta:
                 if elem in listeners:
                     del listeners[elem]
 
+                try:
+                    if getattr(value, "parent") is not None:
+                        # Skip commands if they are apart of a group
```
Comment (Dorukyum) [inline discord/cog.py:150]:
```suggestion
                        # Skip commands if they are a part of a group
```

**Review by Dorukyum — COMMENTED** [review]
Styling :)

File: examples/app_commands/slash_cog_groups.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+from discord.commands.permissions import Permission
+import discord
+from discord.commands.commands import SlashCommandGroup
+from discord.ext import commands
+
+# Set these
+GUILD_ID=0
+OWNER_ID=0
+
+bot = discord.Bot(debug_guild=GUILD_ID, owner_id=OWNER_ID)
```
Comment (RPSMain) [inline examples/app_commands/slash_cog_groups.py:10]: Just a suggestion maybe subclass discord.Bot?

Comment (HexF) [reply examples/app_commands/slash_cog_groups.py:10]: Can you please elaborate?

**Comment by HexF** [comment]
Requesting approval for running workflows.
Should be ready for a merge.

**Pull request merged**